scxml_converter/src/scxml_converter/scxml_entries/utils.py

```diff
@@ -13,26 +13,153 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Collection of various utilities for scxml entries."""
+"""Collection of various utilities for SCXML entries."""
 
-from typing import Any, Dict, Type, MutableSequence
+import re
+from enum import Enum
+from typing import Any, Dict, List, Optional, Type, MutableSequence
 
+from as2fm_common.common import is_array_type, string_to_value
 from scxml_converter.scxml_entries import ScxmlBase
 
+
+PLAIN_SCXML_EVENT_PREFIX: str = "_event."
+
+# Constants related to the conversion of expression from ROS to plain SCXML
+ROS_FIELD_PREFIX: str = "ros_fields__"
+PLAIN_FIELD_EVENT_PREFIX: str = f"{PLAIN_SCXML_EVENT_PREFIX}{ROS_FIELD_PREFIX}"
+
+ROS_EVENT_PREFIXES = [
+    "_msg.",                                                # Topic-related
+    "_req.", "_res.",                                       # Service-related
+    "_goal.", "_feedback.", "_wrapped_result.", "_action."  # Action-related
+]
+
+
 # TODO: add lower and upper bounds depending on the n. of bits used.
 # TODO: add support to uint
 SCXML_DATA_STR_TO_TYPE: Dict[str, Type] = {
-    "bool": bool,
-    "float32": float,
-    "float64": float,
-    "int8": int,
-    "int16": int,
-    "int32": int,
-    "int64": int,
-    "int32[]": MutableSequence[int],  # array.array('i): https://stackoverflow.com/a/67775675
+    "bool":         bool,
+    "float32":      float,
+    "float64":      float,
+    "int8":         int,
+    "int16":        int,
+    "int32":        int,
+    "int64":        int,
+    "int8[]":       MutableSequence[int],    # array('i'): https://stackoverflow.com/a/67775675
+    "int16[]":      MutableSequence[int],
+    "int32[]":      MutableSequence[int],
+    "int64[]":      MutableSequence[int],
+    "float32[]":    MutableSequence[float],  # array('d'): https://stackoverflow.com/a/67775675
+    "float64[]":    MutableSequence[float]
 }
 
 
+# ------------ Expression-conversion functionalities ------------
+class CallbackType(Enum):
+    """Enumeration of the different types of callbacks containing a body."""
+    STATE = 0                   # No callback (e.g. state entry/exit)
+    TRANSITION = 1              # Transition callback
+    ROS_TIMER = 2               # Timer callback
+    ROS_TOPIC = 3               # Topic callback
+    ROS_SERVICE_REQUEST = 4     # Service callback
+    ROS_SERVICE_RESULT = 5      # Service callback
+    ROS_ACTION_GOAL = 6         # Action callback
+    ROS_ACTION_RESULT = 7       # Action callback
+    ROS_ACTION_FEEDBACK = 8     # Action callback
```

Comment:
```suggestion
    STATE = auto()                   # No callback (e.g. state entry/exit)
    TRANSITION = auto()              # Transition callback
    ROS_TIMER = auto()               # Timer callback
    ROS_TOPIC = auto()               # Topic callback
    ROS_SERVICE_REQUEST = auto()     # Service callback
    ROS_SERVICE_RESULT = auto()      # Service callback
    ROS_ACTION_GOAL = auto()         # Action callback
    ROS_ACTION_RESULT = auto()       # Action callback
    ROS_ACTION_FEEDBACK = auto()     # Action callback
```
This way its more maintainable. And you need `from enum import auto`.